### DISCUSSION TITLE
Using Minio

### DISCUSSION THREAD

**Discussion by razor950** [outline/outline#1889] [General]
This is more of a feature that might need to be added, I am using Minio but what am I supposed to set for the aws region because I suppose with minio that isn't being used, could there be a simple environment variable to swap between aws / minio usage when it comes to S3?

**Comment by tommoor** [comment]
Have you tried the default of `us-east-1`?

**Reply by razor950** [reply]
oh I didn't know that was a possible default choice regardless if you are using minio locally. Also does one need to configure CORS on minio's side of things?

**Reply by tommoor** [reply]
There is a `MINIO_REGION_NAME` env arg for setting the region, I'm not too familar beyond that

**Reply by razor950** [reply]
Access to fetch at 'minio url' from origin 'outline url' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.

This issue, I tried a few things but can't seem to fix this at all on my end.

**Reply by tommoor** [reply]
Weird, CORS should be supported and enabled by default in Minio. You'll have to look at the headers on the request to learn more

**Reply by routerino** [reply]
I'm writing up a guide using docker, caddy and minio, it should be published in the next 2-4 days. I can say that I run minio on a separate subdomain (same domain), via https with no CORS issues at all.

**Reply by routerino** [reply]
Blog post about deploying outline with minio is up now: https://blog.gurucomputing.com.au/deploying-outline-with-docker-and-caddy/

**Reply by tommoor** [reply]
Very cool, thanks for putting this together - I'm sure it will be helpful to folks

**Reply by razor950** [reply]
using caddy and deploying minio and outline behind it fixed the issues I had with CORS

**Reply by almereyda** [reply]
Can this be selected as the accepted answer, and thus this discussion be resolved @razor950 ?

**Reply by muuvmuuv** [reply]
@routerino Your post seems to be offline, can you re-upload it somewhere?

**Reply by routerino** [reply]
> @routerino Your post seems to be offline, can you re-upload it somewhere?

Previous guide was out of date, and had some components broken by updates (such as minio).

An updated guide [can be found here](https://blog.gurucomputing.com.au/doing-more-with-docker/deploying-outline-wiki/). There are also related articles to cover the individual components (like nginx proxy manager or keycloak). Other reverse proxies (such as caddy, which I still use) are fine as well.

**Comment by letran3691** [comment]
This is my config when using minio, you can try it.

AWS_ACCESS_KEY_ID=usser
AWS_SECRET_ACCESS_KEY=ACCESS_KEY
#create region in setting console minio
AWS_REGION=ap-southeast-1 
AWS_S3_UPLOAD_BUCKET_URL=https://domain api minio/
AWS_S3_UPLOAD_BUCKET_NAME=outline
AWS_S3_UPLOAD_MAX_SIZE=26214400
AWS_S3_FORCE_PATH_STYLE=true
AWS_S3_ACL=private
